game.py

```diff
@@ -407,6 +412,31 @@ def on_update(self):
                     self.board.overlay.update_unit(self.players[0].units[new_idx])
                 self.map_pos = (clamp(self.board.selected_unit.location[0] - 12, -1, 77),
                                 clamp(self.board.selected_unit.location[1] - 11, -1, 69))
+        elif pyxel.btnp(pyxel.KEY_M):
+            if self.game_started and self.board.overlay.can_iter_settlements_units() and \
+                    len(self.players[0].units) > 0:
+                self.board.overlay.remove_warning_if_possible()
+                if self.board.overlay.is_setl():
+                    self.board.selected_settlement = None
+                    self.board.overlay.toggle_settlement(None, self.players[0])
+                if self.board.selected_unit is None or isinstance(self.board.selected_unit, Heathen):
+                    self.board.selected_unit = self.players[0].units[0]
+                    self.board.overlay.toggle_unit(self.players[0].units[0])
```

Comment:
I believe this part will need to change somewhat, as currently if you press M when you haven't got a unit selected, it will always pick the first unit, regardless of whether it has stamina or not. This could probably be fixed by doing the `filtered_units` calculation from line 428 before this if chain so that you could use that list instead when assigning the selected unit.

Reply:
In fact, I think the usages of `self.players[0].units` on lines 417 and 425 may be problematic as well, since we want the M key to do nothing if there are no units with stamina left.

Reply:
Ah true. Hopefully 2abd84e4d3078ea9dbc776812c5defc338061daf addresses that now.